rmw_connext_shared_cpp/src/qos.cpp

```diff
@@ -85,3 +85,120 @@ get_datawriter_qos(
 
   return true;
 }
+
+template<typename DDSEntityQos>
+bool
+set_entity_qos_from_profile_generic(
+  const rmw_qos_profile_t & qos_profile,
+  DDSEntityQos & entity_qos)
+{
+  // Read properties from the rmw profile
+  switch (qos_profile.history) {
+    case RMW_QOS_POLICY_HISTORY_KEEP_LAST:
+      entity_qos.history.kind = DDS::KEEP_LAST_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_HISTORY_KEEP_ALL:
+      entity_qos.history.kind = DDS::KEEP_ALL_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_HISTORY_SYSTEM_DEFAULT:
+      break;
```

Comment:
I haven't seen where this function is used yet, but we should make sure it's either documented that it expects the `entity_qos` to already be initialized to the "system defaults" or we should change it to explicitly set the default value here. Just want to make sure we're not relying on the compiler to initialize something. Same goes for all the below cases too.

Reply:
This is the existing functionality, just moved out of the header